### ISSUE TITLE
Create a cell that can grab keystrokes and call events on the server

### ISSUE THREAD

**Issue by braxtonmckee** [APrioriInvestments/typed_python#112]
We should be able to accept keystrokes and decide whether to handle them. We should be able to specify an integer priority so that higher priority handlers get to decide first if they want the keystroke.

**Comment by darth-cheney** [comment]
In theory this shouldn't be a problem, but there are a few things we'll want to consider. What does it mean for an arbitrary Cell to listen for keystrokes? On the DOM side, only certain elements can actively have focus (which is required to fire key events). However we can bind key event listeners to the entire body. In that case we need to carefully manage adding/removing handlers as cells come and go.

**Comment by braxtonmckee** [comment]
Yes that's a good point. I was thinking we'd have a single listener on the
DOM, send a message back to a single global cell, and do the routing on the
server side for precisely this reason.  Another thing to consider is how
does this interact with something like the CodeEditor? I think it's
probably OK for native JS components to be in front and have this be the
fallback for any non-consumed keystrokes.

**Comment by darth-cheney** [comment]
> Another thing to consider is how does this interact with something like the CodeEditor? I think it's probably OK for native JS components to be in front and have this be the fallback for any non-consumed keystrokes.
  
The event model in the DOM is that events "bubble up" from the nested children to their ancestors. This is true for most types of events. Any listener can call `event.stopPropagation()` to break the bubbling. But if we have some events listening at the top ancestor level (ie the body) then we just make sure that something like CodeEditor will stop keystroke propagation when it has the focus.

**Comment by darth-cheney** [comment]
Can you provide an example of how you'd envision this Cell being used in practice?
  
I am asking because I have the frontend all set up for it, along with a very basic example Cell. My problem now is that I don't really know how to communicate changes to other cells using purely the declarative structure.
  
Right now I have a `KeyAction` cell whose constructor looks like this:
```
KeyAction('Alt+t', someFunction, ['timeStamp'])
```
Where:
* `'Alt+t'` is the keyCombo that will be registered for the cell on the frontend;
* `someFunction` is a callback that will receive the data from the frontend about the event (this is handled by KeyAction's onMessage);
* `['timeStamp']` is an example of an array of keys that the action should look up on the frontend's event object and return to the server. In this example, all JS events have a timeStamp and that is data that would get sent back to Python when the key event is triggered.
  
I am not quite sure how to make an example in the playground given this setup, because it's unclear to me how to get the `someFunction` to change something in a declarative Cell structure.
  
Any ideas?

**Comment by braxtonmckee** [comment]
So I was thinking of using it like a 'Button' by communicating through a
slot. So for instance

```
def exampleCell():
    lastKeystroke = cells.Slot(None)

    return KeyAction('Alt+t', lambda: lastKeystroke.set('Alt+t')) +
KeyAction('Alt+r', lambda: lastKeystroke.set('Alt+r')) + "You pressed: " +
cells.Subscribed(lambda: lastKeystroke.get())
```
I'm primarily thinking of using this for navigation purposes (e.g.
implementing something like 'ctrl-p' in sublime)

**Comment by darth-cheney** [comment]
Closed by #152
